projects/go-git/build.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash -eu
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+
+go mod download
+go get github.com/AdamKorcz/go-118-fuzz-build/testing
+
+if [ "$SANITIZER" != "coverage" ]; then
+    sed -i '/func (s \*DecoderSuite) TestDecode(/,/^}/ s/^/\/\//' plumbing/format/config/decoder_test.go
+    sed -n '35,$p' plumbing/format/packfile/common_test.go >> plumbing/format/packfile/delta_test.go
+    sed -n '20,53p' plumbing/object/object_test.go >> plumbing/object/tree_test.go
+    sed -i 's|func Test|// func Test|' plumbing/transport/common_test.go
+fi
+
+compile_native_go_fuzzer $(pwd)/internal/revision                       FuzzParser              fuzz_parser
+compile_native_go_fuzzer $(pwd)/plumbing/format/config                  FuzzDecoder             fuzz_decoder
+compile_native_go_fuzzer $(pwd)/plumbing/format/packfile                FuzzPatchDelta          fuzz_patch_delta
+compile_native_go_fuzzer $(pwd)/plumbing/object                         FuzzParseSignedBytes    fuzz_parse_signed_bytes
+compile_native_go_fuzzer $(pwd)/plumbing/object                         FuzzDecode              fuzz_decode
+compile_native_go_fuzzer $(pwd)/plumbing/protocol/packp                 FuzzDecoder             fuzz_decoder
+compile_native_go_fuzzer $(pwd)/plumbing/transport                      FuzzNewEndpoint         fuzz_new_endpoint
```

Comment:
We could use auto discovery instead, so as we add new Fuzz tests in `go-git` they would be automatically picked up:

https://github.com/google/oss-fuzz/blob/master/projects/fluxcd/build.sh#L54

Reply:
> We could use auto discovery instead, so as we add new Fuzz tests in go-git they would be automatically picked up:
> https://github.com/google/oss-fuzz/blob/master/projects/fluxcd/build.sh#L54

It sounds good, but it won't work in go-git due to the [build.sh](https://github.com/google/oss-fuzz/pull/11069/files?diff=unified&w=0#diff-5a1794ba7b100be83ad7b8a7deec8ef88754f3b7d10def72c5e34d35a8c6c588R22-R27) script.

If a new fuzz test is added, I'm not sure it would even compile. There should be some changes in build.sh to ensure the newly added fuzzer works.

Anyway, [go-git](https://github.com/go-git/go-git/pull/858) is planning to use cifuzz to check for build failures and PR fuzzing. Why not switch the build.sh in go-git? For example, i could follow the approach used in go-ethereum's [oss-fuzz.sh](https://github.com/ethereum/go-ethereum/blob/master/oss-fuzz.sh).


Or if https://github.com/google/oss-fuzz/blob/master/projects/fluxcd/build.sh#L54 is really important then there should be some changes in the main `go-git` repo.

There are two options. Which one should I proceed with?

Reply:
Let's take the `go-ethereum`'s approach.

Reply:
Let's break it down.
- https://github.com/go-git/go-git/pull/855
- https://github.com/google/oss-fuzz/pull/11069
- https://github.com/go-git/go-git/pull/858
- New PR in [oss-fuzz](https://github.com/google/oss-fuzz) to remove `build.sh` and use `oss-fuzz.sh`.